app/src/main/java/com/duckduckgo/app/global/AlertingUncaughtExceptionHandler.kt

```diff
@@ -40,4 +48,17 @@ class AlertingUncaughtExceptionHandler(
             originalHandler.uncaughtException(t, originalException)
         }
     }
+
+    /**
+     * Some exceptions happen due to lifecycle issues, such as our sync service being forced to stop.
+     * In such cases, we don't need to alert about the exception as they are exceptions essentially signalling that work was interrupted.
+     * Examples of this would be if the internet was lost during the sync,
+     * or when two or more sync operations are scheduled to run at the same time; one would run and the rest would be interrupted.
+     */
+    private fun shouldIgnore(exception: Throwable?): Boolean {
+        return when (exception) {
+            is UndeliverableException, is InterruptedException, is IOException -> true
```

Comment:
How sure are we about ignoring `IOException`? I can think about many parts of the app that could crash with that error due to bug in our code. For example, working with `File`.

What's our goal here? If it's to ignore `IOException` that we catch and rethrow, we could wrap them into another exception very quick and ignore that one instead. I count 7 places where we do that. But I'm not sure if that's what we want or if I'm missing something else.

Happy to help here and do that change if needed.

Reply:
Good point about `IOException`. Ultimately, the goal is to ignore exceptions which happen due to lifecycle issues such as the bg sync service being instructed to close by the system.

Revisiting the exceptions list, perhaps I over-generalized and should stick with `InterruptedException` and `InterruptedIOException`

<img width="1112" alt="Screen Shot 2020-04-23 at 7 55 08 AM" src="https://user-images.githubusercontent.com/1336281/80068717-fea2d080-8537-11ea-81db-fb92ec9e0aa8.png">

Reply:
Changed this to look for `InterruptedIOException` instead of `IOException`